src/crypto.js

```diff
@@ -1,41 +1,64 @@
 'use strict'
 
 const sha3 = require('js-sha3')
-const murmur3 = require('murmurhash3js')
+const mur = require('murmurhash3js-revisited')
+const sha = require('./sha')
+const { fromNumberTo32BitBuf } = require('./utils')
 
-const utils = require('./utils')
-const sha = require('./crypto-sha1-2')
+const hash = (algorithm) => (data) => {
+  return new Promise((resolve, reject) => {
```

Comment:
Could do the same here as the example in the previous comment, just return value from an `async` function. No need to try/catch and no need for `new Promise`